src/librustc_infer/infer/mod.rs

```diff
@@ -802,11 +1012,24 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         self.skip_leak_check.set(was_skip_leak_check);
 
         let mut inner = self.inner.borrow_mut();
+        let inner = &mut *inner;
+        let InferCtxtInner {
+            type_variables,
+            const_unification_table,
+            int_unification_table,
+            float_unification_table,
+            ..
+        } = inner;
+        inner.undo_log.rollback_to(
+            &mut RollbackView {
```

Comment:
Why do we implement `Rollback` for `RollbackView` and not `InferCtxtInner`?

Reply:
Since the `undo_log` is part of `InferCtxtInner` I had to separate the fields to implement `Snapshots`. There doesn't seem to be a point to implement `Snapshots` explicitly though so I could implement the equivalent methods without being restricted to the trait and have it work without `RollbackView`